File: Grind75/C++/longestSubstringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int frequency[128] = {0};
        int i = 0, j = 0, maxLen = 0;
        while (j < s.length()) {
            if (frequency[int(s[j])] == 0) {
                frequency[int(s[j])] = 1;
                j++;
            } else {
                while (frequency[int(s[j])] != 0) {
                    frequency[int(s[i])] = 0;
                    i++;
                }
            }
            maxLen = max(maxLen,(j - i));
        }
        return maxLen;
    }
};

File: longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxLen = 0
        freq = [0]*128
        i,j = 0,0
        while j < len(s):
            if freq[ord(s[j])] == 0:
                freq[ord(s[j])] += 1
            else:
                while freq[ord(s[j])] != 0:
                    freq[ord(s[i])] -= 1
                    i += 1
                freq[ord(s[j])] = 1
            maxLen = max(maxLen, (j-i)+1)
            j += 1
        return maxLen
